samcli/local/docker/lambda_build_container.py

```diff
@@ -104,8 +130,23 @@ def __init__(  # pylint: disable=too-many-locals
             additional_volumes=additional_volumes,
             entrypoint=entry,
             env_vars=env_vars,
+            mount_with_write=mount_with_write,
         )
 
+    def make_host_tmp_dirs(self):
+        """
+        Make tmp dirs on host
+        """
+        if self._host_tmp_base_dir and not os.path.exists(self._host_tmp_base_dir):
```

Comment:
So we won't create it if it already exists, but then we'll delete it afterwards? Can we not use a common path like /tmp/samcli ? A user might actually use that for something. Can we just use a GUID or something for the path? This is on the host machine, right? It doesn't have to match the path inside the container, right?